ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -98,11 +113,25 @@ internal constructor(
     val currentTask = taskViewModels.value!![currentPosition.value!!]
     val validationError = currentTask.validate()
     if (validationError == null) {
-      // TODO(#1146): Handle the scenario when the user clicks next on the last Task. This will
-      //  include persisting the list of responses to the database
+      responses[currentTask.task] = currentTask.taskData.value?.orElse(null)
+
+      // TODO(jsunde): Test this behavior
+      if (currentPosition.value!! == taskCount.value!! - 1) {
```

Comment:
For some reason `taskCount.value` is null in my DataCollectionFragmentTests.

Any ideas why that might be? I don't really understand it because `jobName.value` returns "Job" and the taskCount LiveData is set up in pretty much the same way

Reply:
Resolved offline